src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -257,6 +257,13 @@ def _detect_performance_problems(data: Event, sdk_span: Any) -> List[Performance
     event_id = data.get("event_id", None)
     project_id = data.get("project")
 
+    try:
+        project = Project.objects.get_from_cache(id=project_id)
+        organization = Organization.objects.get_from_cache(id=project.organization_id)
+    except Project.DoesNotExist:
+        project = None
+        organization = None
+
```

Comment:
We should already have the project available in the caller, let's just pass it in here? It will have the organization precached as well. Saves us making extra io calls for no reason.

Reply:
Done in [fd9b75d](https://github.com/getsentry/sentry/pull/42777/commits/fd9b75dba68bdb7c823d6e305289d1a9f0d60392)!